transport.go

```diff
@@ -167,6 +172,14 @@ func (t *Transport) Repositories() ([]github.Repository, error) {
 	return t.token.Repositories, nil
 }
 
+// Expiry returns a transport token's expiration time and current expiration status.
+func (t *Transport) Expiry() (expiresAt time.Time, expired bool, err error) {
+	if t.token == nil {
+		return time.Time{}, true, errors.New("Expiry() = unknown, err: nil token")
+	}
+	return t.token.ExpiresAt, t.token.isExpired(), nil
```

Comment:
The token expiration time may not always line up with whether ghinstallation considers the token to be expired - we add a buffer and refresh the token early to avoid potential races due to unexpected delays. It might look like a bug if the expire_at time > now but the func says it's expired.

To brainstorm some other ideas - what if we add a func that returns a https://pkg.go.dev/golang.org/x/oauth2#Token? this way we can separate out what ghinstallation considers expired vs if the token itself is expired.

e.g.

```go
token := t.OAuth2Token()  // name tbd, we just can't use Token because that's already taken
expiresAt, expired := token.Expiry, token.Valid()
```

we could also do some refactoring to implement the [TokenSource interface](https://pkg.go.dev/golang.org/x/oauth2#TokenSource) 🤔 (this would need to be a new type though)

Reply:
> The token expiration time may not always line up with whether ghinstallation considers the token to be expired - we add a buffer and refresh the token early to avoid potential races due to unexpected delays. It might look like a bug if the expire_at time > now but the func says it's expired.

This was really why I factored out and returned the "expired" flag. I guess my intention was to indicate that, if we ran `Token()` right now  (sans races), would we generate a new token.

> what if we add a func that returns a https://pkg.go.dev/golang.org/x/oauth2#Token? this way we can separate out what ghinstallation considers expired vs if the token itself is expired.

I don't mind this approach, but I would suggest, at least for now, a simpler approach: instead of returning "expired", perhaps I return a flag to indicate that it will be refreshed, e.g. "refreshPending" or something like that?

Reply:
We could return the time that we'll refresh the token `at.ExpiresAt.Add(-time.Minute)`, then if we change the grace period then the client will still know when we'll refresh. The user could then log this and interpret the results themselves.

We could return `at.ExpiresAt` instead of whether it's expired too. But this is when the token itself expires, not when we'll refresh. This doesn't seem helpful though.

Reply:
Yeah, I like this approach. I'll return expiresAt for reference as well as this refreshTime that you're describing. Thanks!

Reply:
OK, I've tried this out in https://github.com/bradleyfalzon/ghinstallation/pull/85/commits/01726501c9d0ab2b6538d884eac1e40c69211087

Reply:
👍 I'm good with this compromise. Thanks!

(I'll give @bradleyfalzon a chance to look at this, otherwise I'll merge on Monday)